router/batchrouter/worker.go

```diff
@@ -28,259 +23,223 @@ func newWorker(partition string, logger logger.Logger, brt *Handle) *worker {
 		partition: partition,
 		logger:    logger,
 		brt:       brt,
+		cb: circuitbreaker.NewCircuitBreaker(
+			partition,
+			circuitbreaker.WithMaxRequests(1),
+			circuitbreaker.WithTimeout(brt.conf.GetDurationVar(10, time.Second, "BatchRouter.timeout", "BatchRouter."+brt.destType+".uploadFreq", "BatchRouter.uploadFreq")),
+			circuitbreaker.WithConsecutiveFailures(brt.conf.GetIntVar(3, 1, "BatchRouter.maxConsecutiveFailures", "BatchRouter."+brt.destType+".maxConsecutiveFailures")),
+			circuitbreaker.WithLogger(logger),
+		),
 	}
+	w.pw = NewPartitionWorker(logger, partition, brt, w.cb)
+	w.pw.Start()
 	return w
 }
 
 type worker struct {
 	partition string
 	logger    logger.Logger
 	brt       *Handle
+	cb        circuitbreaker.CircuitBreaker
+	pw        *PartitionWorker
 }
 
 // Work retrieves jobs from batch router for the worker's partition and processes them,
 // grouped by destination and in parallel.
 // The function returns when processing completes and the return value is true if at least 1 job was processed,
 // false otherwise.
 func (w *worker) Work() bool {
+	// Check if circuit breaker is open before doing any work
+	if w.cb.IsOpen() {
+		w.logger.Warnn("Circuit breaker is open, skipping job processing",
+			logger.NewStringField("partition", w.partition))
+		return false
+	}
+
 	brt := w.brt
 	workerJobs := brt.getWorkerJobs(w.partition)
 	if len(workerJobs) == 0 {
 		return false
 	}
 
-	var jobsWg sync.WaitGroup
-	jobsWg.Add(len(workerJobs))
 	for _, workerJob := range workerJobs {
-		w.processJobAsync(&jobsWg, workerJob)
+		w.routeJobsToBuffer(workerJob)
 	}
-	jobsWg.Wait()
 	return true
 }
 
-// processJobAsync spawns a goroutine and processes the destination's jobs. The provided wait group is notified when the goroutine completes.
-func (w *worker) processJobAsync(jobsWg *sync.WaitGroup, destinationJobs *DestinationJobs) {
+// routeJobsToBuffer sends jobs to appropriate channels in the job buffer
+func (w *worker) routeJobsToBuffer(destinationJobs *DestinationJobs) {
```

Comment:
This doesn't realy do any kind of routing, it merely pushes jobs to a channel
```suggestion
func (w *worker) scheduleJobs(destinationJobs *DestinationJobs) {
```